lib/app/modules/bulk_data/bulk_data_patient_export_sequence.rb

```diff
@@ -59,13 +74,68 @@ def export_kick_off(klass)
           versions :stu3
         end
 
-        reply = export_kick_off('Patient')
+        assert_export_kick_off('Patient')
+      end
 
-        assert_response_accepted(reply)
+      test 'Server shall return "202 Accepted" or "200 OK"' do
+        metadata do
+          id '03'
+          link 'https://build.fhir.org/ig/HL7/bulk-data/export/index.html#bulk-data-status-request'
+          desc %(
+          )
+          versions :stu3
+        end
 
-        @content_location = reply.response[:headers]['content-location']
+        assert_export_status(@content_location)
+      end
+
+      private
+
+      def export_kick_off(klass)
+        headers = { accept: 'application/fhir+json', prefer: 'respond-async' }
+
+        url = ''
+        url += "/#{klass}" if klass.present?
+        url += '/$export'
+
+        @client.get(url, @client.fhir_headers(headers))
+      end
+
+      def export_status_check(url, timeout)
+        wait_time = 1
+        reply = nil
+        headers = { accept: 'application/json' }
+        start = Time.now
 
-        assert @content_location.present?, 'Server response must have "Content-Location" in header for $export request'
+        loop do
+          reply = @client.get(url, @client.fhir_headers(headers))
+
+          wait_time = get_wait_time(wait_time, reply)
+          seconds_used = Time.now - start + wait_time
+
```

Comment:
Regarding the timeouts --would it be easier to at this point do

```skip "Server took more than #{timeout} seconds to process the request."  if seconds_used > timeout``` 

And then get rid of the other timeout skipping code earlier?  It is hard for me to follow the logic between timeouts and skipping over there, because it is not obvious why you would skip when `status == 202`.  I trust that it works, it just that it will take me some time to figure why it would work.

Reply:
export_status_check function just check status. It does not handle any assert/skip logic. The logic here is that if the status is still 202 and pass the timeout threshold, export_status_check stops next status check and returns status 202 to caller. The caller, assert_export_status handles assert/logic based on the returned status code. If the code is 200, the status check completed. If the code is 202, the status check timed out. Otherwise, there is server error. 
The function export_status_check may finally merged into fhirclient module as common bulk data export function